examples/platform/silabs/provision/ProvisionStorageDefault.cpp

```diff
@@ -70,14 +70,29 @@ CHIP_ERROR ErasePage(uint32_t addr)
     return chip::DeviceLayer::Silabs::GetPlatform().FlashErasePage(addr);
 }
 
-CHIP_ERROR WritePage(uint32_t addr, const uint8_t * data, size_t size)
+size_t RoundNearest(size_t n, size_t multiple)
 {
-    return chip::DeviceLayer::Silabs::GetPlatform().FlashWritePage(addr, data, size);
+    return (n % multiple) > 0 ? n + (multiple - n % multiple) : n;
 }
 
-size_t RoundNearest(size_t n, size_t multiple)
+/**
+ * Writes "size" bytes to the flash page. The data is padded with 0xff
+ * up to the nearest 32-bit boundary. The "max" argument is used to ensure
+ * that the padding won't exceed the limits of the buffer.
+ */
+CHIP_ERROR WritePage(uint32_t addr, const uint8_t * data, size_t size, size_t max)
 {
-    return (n % multiple) > 0 ? n + (multiple - n % multiple) : n;
+    // The flash driver fails if the size is not a multiple of 4 (32-bits)
+    size_t size_32 = RoundNearest(size, 4);
+    // If the input data is smaller than the 32-bit size, pad the buffer with "0xff"
+    if (size_32 != size)
+    {
+        uint8_t * p = (uint8_t *) data;
```

Comment:
Why is casting away const ok here?  What if the caller actually passes a pointer to readonly memory?

Seems like if we write to the buffer, the passed-in bit needs to not be const.

Reply:
This is the result of a desperate attempt to save RAM, but it is definitely wrong. It is currently working because I know the location and size of the original buffer, but this practice is dangerous, and should be avoided. I'll find a better workaround.

Reply:
Again: it seems like this would be fine if the API were not declared as const.